openedx/frontend-app-course-authoring
# feat: create library form [FC-0059]

## Description
This PR adds a Create Library form to course-authoring.

![image](https://github.com/openedx/frontend-app-course-authoring/assets/849463/e8bea0f7-fb0a-4d06-8f8c-bb41541e7473)

## More Info
- Closes: https://github.com/openedx/frontend-app-course-authoring/issues/1036
- Depends on:
  - https://github.com/openedx/frontend-app-course-authoring/pull/1076
  - https://github.com/openedx/edx-platform/pull/34934

## Testing Instructions
- Run your local stack on this branch for course authoring and https://github.com/openedx/edx-platform/pull/34934 for edx-platform
- Make sure you have the studio.library_authoring_mfe waffle flag **DISABLED**
- Go to the Studio Home and go to the "Libraries" tab
- Click on the "+ New library" button
- You should be redirected to the new library form
- Fill the fields trying some invalid data to check validation
- Fill the fields correctly and hit "Create".
- A new library should be created, and you should be redirected to its Home
- Go back to Studio Home and create a new library using the same data (to force a duplicate ID)
- You should see an error message after submitting

---
Private ref: [FAL-3768](https://tasks.opencraft.com/browse/FAL-3768)
